googletest/include/gtest/internal/gtest-port.h

```diff
@@ -396,7 +396,7 @@
 #  include <io.h>
 # endif
 // In order to avoid having to include <windows.h>, use forward declaration
-#if GTEST_OS_WINDOWS_MINGW
+#if GTEST_OS_WINDOWS_MINGW && !defined(__MINGW64_VERSION_MAJOR)
```

Comment:
I can confirm that MinGW 4.8.2 g++ does not define this symbol.
The only symbols defined by that compiler with "MINGW" in its name are:
* `__MINGW32__` for the 32 bit compiler.
* both `__MINGW32__` and `__MINGW64__` for the 64 bit compiler